### PR TITLE
TODO comment reduction

### PR DESCRIPTION
**Pull request by Dweblenod**
Working towards reducing the number of TODO comments in code by handling simpler improvement tasks.

### CODE REVIEW

**Comment by Dweblenod** [comment]
Holding off on making it a full PR instead of draft after what should be a short but undecided length of time

File: src/main/java/com/mraof/minestuck/entity/consort/MessageType.java
Comment (Dweblenod) [inline src/main/java/com/mraof/minestuck/entity/consort/MessageType.java:1063]: there is no longer an index to utilize for this nbt so it has been removed. Overall let me know if there is anything I cut out from the original code that you would like to be returned

File: src/main/java/com/mraof/minestuck/skaianet/SkaianetHandler.java
Comment (Dweblenod) [inline src/main/java/com/mraof/minestuck/skaianet/SkaianetHandler.java:346]: I do not feel like I can make an adequate judgement on this. This comment continues the TODO in ComputerReference, will remove it if someone has a comment to make in this PR

File: src/main/java/com/mraof/minestuck/entity/consort/MessageType.java
Comment (kirderf1) [reply src/main/java/com/mraof/minestuck/entity/consort/MessageType.java:1063]: The index was originally used to identify which of the possible items in the list that the message type has targeted. As far as I can tell, you could use the resource name to identify which item in the tag that is targeted.

Also, some of the functionality seems to have been lost. I'd suggest checking out the javadoc on the constructor if you haven't already.

File: src/main/java/com/mraof/minestuck/skaianet/SkaianetHandler.java
Comment (kirderf1) [reply src/main/java/com/mraof/minestuck/skaianet/SkaianetHandler.java:346]: Reworking `checkData()` would proably be out of scope for this PR, so I suggest leaving it be for now.

Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/skaianet/SkaianetHandler.java:346]: Would you like me to remove the TODO here or is it appropriate to keep?

Comment (kirderf1) [reply src/main/java/com/mraof/minestuck/skaianet/SkaianetHandler.java:346]: As long as there's already a todo adressing `checkData()` elsewhere, I'd say remove this one

File: src/main/java/com/mraof/minestuck/entity/consort/MessageType.java
Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/entity/consort/MessageType.java:1062]: Switched the index to instead be a string of the registry name. Looking at nbt and playing out a consort interaction, the functionality appears to be restored. The text does not repeat and the consort only asks for a specific item if it was being held.

`MessageData: {380df991-f603-344c-a090-369bad2a924a: {hungry.item: {id: "minecraft:cookie", Count: 1b}, hungry: 0, currentMessage: "hungry.ask_food"}}`

File: src/main/java/com/mraof/minestuck/entity/consort/MessageType.java
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/entity/consort/MessageType.java:1076]: This should put a string to match the changes above.

Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/entity/consort/MessageType.java:1076]: oops missed that

File: src/main/java/com/mraof/minestuck/client/gui/StoneTabletScreen.java
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/client/gui/StoneTabletScreen.java:108]: The comment is insufficient to identify or recreate the issue. Where is the full crash log? Or is this just meant as a reminder for yourself before this PR gets merged?

**Comment by kirderf1** [comment]
I think the PR is close to an appropriate size to finish up and be merged. What are your thoughts on it?

File: src/main/java/com/mraof/minestuck/client/gui/StoneTabletScreen.java
Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/client/gui/StoneTabletScreen.java:108]: I shared the context in the server, so since the problem has been noted there I will remove it here

File: src/main/java/com/mraof/minestuck/world/gen/feature/MSPlacedFeatures.java
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/world/gen/feature/MSPlacedFeatures.java:151]: Check out what that placement modifier helper function does. It already takes care of some of those modifiers.

File: src/main/java/com/mraof/minestuck/world/biome/LandBiomeHolder.java
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/world/biome/LandBiomeHolder.java:131]: Lands shouldn't use these ore features that are defined for the overworld. They should be using the ore block and ground type provided by the `StructureBlockRegistry`. Please revert the changes to this class.

File: src/main/java/com/mraof/minestuck/world/gen/feature/MSPlacedFeatures.java
Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/world/gen/feature/MSPlacedFeatures.java]: Let me know if I missed any other modifiers already handled by worldGenModifiers()

**Review by kirderf1 — APPROVED** [review]

**Pull request merged**
